training.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import gym
+from pygame.constants import K_ESCAPE, KEYDOWN
+import gym_game
+import pygame as pg
+import os
+import sys
+import keyboard
+
+import torch
+import torch.optim as optim
+
+import random
+from agent import DirectPolicyAgent
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+device = 'cpu'
+agent = DirectPolicyAgent(device)
+agent.to(device)
+
+pg.init()
+
+# Parameters
+generations = 10
+episodes_per_gen = 10000 # Episodes before new generation
+batch_size = 10 #Episodes before param update
+learning_rate = 0.0001 # Learning rate
+decay_rate = 0.99 # Weight decay for Adam optimizer
+
+# Optimizer
+optimizer = optim.Adam(agent.parameters(), lr=learning_rate, weight_decay=decay_rate)
+
+# Environment
+env = gym.make('ConnectFour-v0')
+
+s = env.reset()
+env.configurePlayer(random.choice([-1,1]))
+
+# Main training loop
+# for gen in range(generations):
+#     model_save_path = 'AgentParameters/agent_params'+gen+'.pkl'
```

Comment:
Check out docs for `torch_save()` if you don't know about these. I think they would be the easiest way of saving model parameters (as well as perhaps other data)

Reply:
See second section in #10.